colorls.rb

```diff
@@ -124,21 +172,28 @@ def options(content)
   end
 end
 
-args = *ARGV
+args         = *ARGV
+report       = false
+sort         = false
+one_per_line = false
 
-if args.include?('--report') || args.include?('-r')
-  report = true
-  args -= %w[--report -r]
-else
-  report = false
+args.each do |arg|
+  report       = true if ['--report', '-r'].include?(arg)
+  one_per_line = true if arg == '-1'
+
+  match = arg.match(/^--sort=?(.*)?$/)
+
+  if match
+    sort = match.captures[0] == 'dirs-first' ? 'dirs-first' : true
+  end
 end
```

Comment:
Can the flag for `--sort=` be changed to : 
`-sf` or `--sort-files` : Show files first
`-sd` or `--sort-dirs` : Show dirs first

An extension of this could be
`-f` or `--files` : Show only files
`-d` or `--dirs` : Show only dirs